src/tests/ftest/util/ior_test_base.py

```diff
@@ -138,8 +138,9 @@ def run_ior_with_pool(self, intercept=None, display_space=True, test_file_suffix
                 This will enable dfuse (xattr) working directory which is
                 needed to run vol connector for DAOS. Default is None.
             timeout (int, optional): command timeout. Defaults to None.
-            fail_on_warning (bool, optional): Controls whether the test
-                should fail if a 'WARNING' is found. Default is False.
+            fail_on_warning (bool/callable, optional): Controls test behavior when a 'WARNING' is
+                found. If True, call self.fail. If False, call self.log.warn. If callable, call it.
+                Default is False.
```

Comment:
Ideally, this parameter should be rearchitected, but I was trying to avoid updating all callers